backend/src/db/migrations/20240429175301_fix-db-reference-for-groups-and-project-memberships.ts

```diff
@@ -0,0 +1,78 @@
+import { Knex } from "knex";
+
+import { TableName } from "../schemas";
+
+export async function up(knex: Knex): Promise<void> {
+  // --- SECRET APPROVALS START ----
+
+  const hasNewColumn = await knex.schema.hasColumn(TableName.SecretApprovalRequestReviewer, "memberUserId");
+
+  if (!hasNewColumn) {
+    await knex.schema.alterTable(TableName.ProjectUserAdditionalPrivilege, (t) => {
+      // add new "groupProjectId" column
+      t.string("groupProjectId").nullable();
+      t.foreign("groupProjectId").references("id").inTable(TableName.Project).onDelete("CASCADE");
+    });
+
+    await knex.schema.alterTable(TableName.SecretApprovalPolicyApprover, (t) => {
+      t.uuid("approverId").nullable().alter();
+
+      // add new "approverUserId" column
+      t.uuid("approverUserId").nullable();
+      t.foreign("approverUserId").references("id").inTable(TableName.Users).onDelete("CASCADE");
+    });
+
+    await knex.schema.alterTable(TableName.SecretApprovalRequest, (t) => {
```

Comment:
I was testing identity for secret replication and issue is then i would need to add all these fields again . May be instead we do it like permission there is non nullable   - `type"which says whether its user or identity and keep rest as generic like before

Then do a conditional join

Reply:
Kindly make the migration file idempotent